src/algorithms/calorimetry/ImagingTopoCluster.cc

```diff
@@ -153,8 +176,21 @@ void ImagingTopoCluster::process(const Input& input, const Output& output) const
     idx = indices.erase(idx); // takes role of idx++
   }
   debug("found {} potential clusters (groups of hits)", groups.size());
+  // for (std::size_t i = 0; i < groups.size(); ++i) {
+  //   debug("group {}: {} hits", i, groups[i].size());
+  // }
   for (std::size_t i = 0; i < groups.size(); ++i) {
+    debug("");
+    debug("");
     debug("group {}: {} hits", i, groups[i].size());
+    for (auto idx : groups[i]) {
+      const auto& hit = (*hits)[idx];
+      debug("  hit {} -> energy = {:.6f}, layer = {}, sector = {}, local = ({:.2f}, {:.2f}, "
```

Comment:
There's a level that is even more noisy than `debug`, and that's `trace`. That's intended for output at the individual hit level in clustering algorithms like this.
```suggestion
      trace("  hit {} -> energy = {:.6f}, layer = {}, sector = {}, local = ({:.2f}, {:.2f}, "
```